### PR TITLE
Tracing: Always use Once to initialize tracing_subscriber

### PR DESCRIPTION
**Pull request by youngjoon-lee**
## 1. What does this PR implement?
I've tried to use `Tracing` service for some integration tests that runs multiple Overwatch instance(?) that contains the `Tracing` service. But, the instances panic-ed because the tracing_subscriber was initialized more than once. 
I found that the `Tracing` service uses `Once` to initialize the tracing_subscriber only when `#[cfg(test)]` is enabled. But, the `test` configuration flag is enabled only when running unit tests. It is not enabled for integration tests, if I understand it correctly.

So, in this PR, I changed the `Tracing` service to always use `Once` regardless of `#[cfg(test)]`, because we anyway should initialize the tracing_subscriber only once. 

Please let me know if this is not correct, or if there is any better way. I've tested that the integration test with multiple Overwatch instances work well. Also, the standalone `nomos-node` prints logs well.

## 2. Does the code have enough context to be clearly understood?

Yes

## 3. Who are the specification authors and who is accountable for this PR?

@youngjoon-lee @bacv 
## 4. Is the specification accurate and complete?

Yes

## 5. Does the implementation introduce changes in the specification?

No

## Checklist

> [!WARNING]  
> Do not merge the PR if any of the following is missing:

* [ ] 1. Description added.
* [ ] 2. Context and links to Specification document(s) added.
* [ ] 3. Main contact(s) (developers and specification authors) added
* [ ] 4. Implementation and Specification are 100% in sync including changes. This is critical.
* [ ] 5. Link PR to a specific milestone.

### CODE REVIEW

**Review by bacv — APPROVED** [review]
> So, in this PR, I changed the Tracing service to always use Once regardless of #[cfg(test)]

I think this change makes sense. But it is still not very clear why couple of Overwatch instances in the same process, but with `#[cfg(test)]` flag failed to initialize it once?

**Comment by bacv** [comment]
> But, the test configuration flag is enabled only when running unit tests.

Which command are you using to run tests?

**Comment by youngjoon-lee** [comment]
> > But, the test configuration flag is enabled only when running unit tests.
> 
> Which command are you using to run tests?

I ran:
```
cargo test -p <crate-name> —tests
```
I wrote a simple test by creating `nomos-services/mix/tests/` directory. I didn’t commit this to the repo because this was just a temporary test to check something on my side, and I think you’re preparing something similar. 
Maybe I’m running tests in a wrong way. Let me also double check about the config flag.

**Comment by danielSanchezQ** [comment]
I do not get the issue here.

If using `Once` it should work for both single process (shared memory) and multiprocess (non-shared memory).

**Comment by youngjoon-lee** [comment]
> I do not get the issue here.
> 
> If using `Once` it should work for both single process (shared memory) and multiprocess (non-shared memory).

Yes, but the issue was that `Once` wasn't used when running integration tests (not unit tests).

**Comment by youngjoon-lee** [comment]
> Aaah, that should not matter. In integration tests we are using independent instances (processes) for each node (iirc). So the issue would be something else entirely 🤔

Ah, let me rephrase my description. Sorry for confusion. @danielSanchezQ @bacv 
I'm not taking about the integration tests in `root/tests/`. I wrote a test under the `nomos-services/mix/tests/`. I didn't open a PR for this test yet. I thought I mentioned this in the PR description, but I just found that I didn't :)
https://github.com/logos-co/nomos-node/blob/mix-conn-monitor-intergration-test/nomos-services/mix/tests/frequency.rs
I call this integration test (as introduced in https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests), but yeah it's confusing.

In this test, we have the [`TestNode`](https://github.com/logos-co/nomos-node/blob/78b8cc585e1864512fc8321e12f59672577663c2/nomos-services/mix/tests/frequency.rs#L94) that has `Tracing` service, and we run multiple `TestNode`s using Overwatch in one process. However, because `#[cfg(test)]` is not enabled when running integration(?) tests, the `Once` is not used.

It may be better to re-open this PR when I open a PR for this test.

**Comment by youngjoon-lee** [comment]
Closing this since nobody feels a need of this now.I will reopen this when I open PRs for testing blend modules.
